jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -228,15 +228,58 @@ public static Booster train(
             break;
           }
         }
-        if (Rabit.getRank() == 0) {
-          Rabit.trackerPrint(evalInfo + '\n');
+        if (Rabit.getRank() == 0 && shouldPrint(params, iter)) {
+          if (shouldPrint(params, iter)){
+            Rabit.trackerPrint(evalInfo + '\n');
+          }
         }
       }
       booster.saveRabitCheckpoint();
     }
     return booster;
   }
 
+  static Integer tryGetIntFromObject(Object o) {
```

Comment:
this should be given private accessibility

Reply:
Agreed, I've made this change.